### PR TITLE
ZBUG-1578: added castNicknameAsString function in normalize-otherAttribute-contact.ts file to handle array values of nickname.

### PR DESCRIPTION
**Pull request by madhavdhuppe**

### CODE REVIEW

File: src/batch-client/index.ts [hunk 1]
```diff
@@ -796,7 +797,13 @@ export class ZimbraBatchClient {
 				},
 				...rest
 			}
-		}).then(res => res.cn.map((contact: any) => normalize(Contact)(contact)));
+		}).then(res =>
+			res.cn.map((cn: any) => {
+				const contact = castNicknameAsString(cn);
```
Comment (silentsakky) [inline src/batch-client/index.ts:802]: all other contact related functions are calling normalizeOtherAttr, why getContact is not doing it? ideally it should also do same as others

File: src/utils/normalize-otherAttribute-contact.ts [hunk 2]
```diff
@@ -103,8 +103,9 @@ export function createContactBody(data: any) {
 }
 
 export function normalizeOtherAttr(data: any) {
-	return data.map((contact: any) => {
+	return data.map((cn: any) => {
```
Comment (silentsakky) [inline src/utils/normalize-otherAttribute-contact.ts:106]: contact was better

File: src/batch-client/index.ts (hunk 1 above)
Comment (madhavdhuppe) [reply src/batch-client/index.ts:802]: ok checking i didn’t updated the code

File: src/utils/normalize-otherAttribute-contact.ts (hunk 2 above)
Comment (madhavdhuppe) [reply src/utils/normalize-otherAttribute-contact.ts]: updated

File: src/utils/normalize-otherAttribute-contact.ts [hunk 3]
```diff
@@ -106,6 +106,19 @@ export function normalizeOtherAttr(data: any) {
...
+
+		if (castValueOfNickname) {
+			// Ideally we shouldn't get the nickname as type of Array as we are not supporting the multiple nicknames
+			// this only happens when user import PST using legacy by using tool name Zimbra PST Migration Wizard
+			// So, here if nickname is array then use its first element as nickname
+			castValueOfNickname = Array.isArray(castValueOfNickname)
+				? castValueOfNickname[0]
+				: castValueOfNickname;
+
+			delete contact._attrs['nickname'];
```
Comment (silentsakky) [inline src/utils/normalize-otherAttribute-contact.ts:119]: you can just do in place replace
```
contact._attrs['nickname'] = Array.isArray(castValueOfNickname)
				? castValueOfNickname[0]
				: castValueOfNickname;
```

Comment (madhavdhuppe) [reply src/utils/normalize-otherAttribute-contact.ts:119]: sure

**Pull request merged**